lib/api/index.ts

```diff
@@ -0,0 +1,50 @@
+type LPAResponse = {
+  entities: LPAEntity[];
+};
+
+type LPAEntity = {
+  "entry-date": string;
+  "start-date": string;
+  "end-date": string;
+  entity: number;
+  name: string;
+  dataset: string;
+  typology: string;
+  reference: string;
+  prefix: string;
+  "organisation-entity": string;
+  geometry: string;
+  point: string;
+  region: string;
+  website: string;
+  wikidata: string;
+  "billing-authority": string;
+  "local-authority-type": string;
+  "parliament-thesaurus": string;
+  "addressbase-custodian": string;
+  "statistical-geography": string;
+  "local-resilience-forum": string;
+  "opendatacommunities-uri": string;
+  "local-authority-district": string;
+};
+
+export const fetchLPAs = (): Promise<LPAResponse> => {
+  return new Promise((resolve, reject) => {
+    fetch(
+      "https://www.planning.data.gov.uk/entity.json?dataset=local-authority&limit=400"
+    )
+      .then((response) => {
+        response
+          .json()
+          .then((data) => {
+            resolve(data);
+          })
+          .catch((error) => {
+            reject(new Error(error));
+          });
+      })
+      .catch((error) => {
+        reject(new Error(error));
+      });
+  });
+};
```

Comment:
We can make this more succinct and it's also worth noting that I don't think the way this is throwing the errors is correct. The `Error` constructor only accepts a `string` parameter, which `error` probably wouldn't be (although the actual type throw is `unknown`. We probably want something along the lines of this:

```suggestion
export const fetchLPAs = async (): Promise<LPAResponse> => {
  try {
    const response = await fetch(
      "https://www.planning.data.gov.uk/entity.json?dataset=local-authority&limit=400"
    );
    return response.json();
  } catch (error) {
    if (error instanceof Error) {
      throw new Error("Failed to fetch LPAs: " + error.message);
    }
    
    throw new Error("Failed to fetch LPAs: " + JSON.stringify(error));
  }
};
```